homeassistant/components/rachio/coordinator.py

```diff
@@ -54,3 +66,55 @@ async def _async_update_data(self) -> dict[str, Any]:
         except Timeout as err:
             raise UpdateFailed(f"Could not connect to the Rachio API: {err}") from err
         return {valve[KEY_ID]: valve for valve in data[1][KEY_VALVES]}
+
+
+class RachioScheduleUpdateCoordinator(DataUpdateCoordinator[list[dict[str, Any]]]):
+    """Coordinator for fetching hose timer schedules."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        rachio: Rachio,
+        base_station,
+    ) -> None:
+        """Initialize a Rachio schedule coordinator."""
+        self.hass = hass
+        self.rachio = rachio
+        self.base_station = base_station
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=f"{DOMAIN} schedule update coordinator",
+            update_interval=timedelta(minutes=30),
```

Comment:
This is pretty conservative as well, with 2 base stations I’m only hitting around 1000 API calls/day, but I don’t see any reason to go less than 15 min intervals.